### PR TITLE
WT-8965 __wt_rec_time_window_clear_obsolete false positives

### PR DESCRIPTION
**Pull request by sauclovian-wt**
Avoid causing extra work in __wt_rec_time_window_clear_obsolete.

If the start is already empty, it'll be globally visible, but don't clear it again; not because writing zeros back to the time window is
bad but because setting WT_CELL_UNPACK_TIME_WINDOW_CLEARED can cause row-store reconciliation to repack the value cell when it doesn't actually need to.

### CODE REVIEW

**Comment by lukech** [comment]
autobuild: Can one of the admins verify this patch?

File: src/include/reconcile_inline.h [hunk 1]
```diff
@@ -502,9 +502,12 @@ __wt_rec_time_window_clear_obsolete(
     if (!tw->prepare && !F_ISSET(S2C(session), WT_CONN_IN_MEMORY)) {
         /*
          * Check if the start of the time window is globally visible, and if so remove unnecessary
-         * values.
+         * values. Avoid doing anything if the values are already clear, as setting the cleared flag
+         * can cause row-store reconciliation to do extra work repacking values.
          */
-        if (WT_REC_TW_START_VISIBLE_ALL(r, tw)) {
```
Comment (kommiharibabu) [inline src/include/reconcile_inline.h:507]: Does the check above WT_TIME_WINDOW_HAS_START do something similar? Isn't it?

Comment (sauclovian-wt) [reply src/include/reconcile_inline.h:507]: Hmm. I guess so -- not sure how I missed that.

I think that means this change is useless...

Comment (kommiharibabu) [reply src/include/reconcile_inline.h:507]: Yes.

Comment (kommiharibabu) [reply src/include/reconcile_inline.h:507]: Earlier there was another macro to check whether the time window exists or not. As we are clearing only the start time window here, I changed the macro. Maybe you might have checked the code before I did that change.

Comment (sauclovian-wt) [reply src/include/reconcile_inline.h:507]: No, I think I just missed it. Happens; that's why we do reviews :-)

thanks for catching it

Comment (sauclovian-wt) [reply src/include/reconcile_inline.h:507]: (and sorry about the noise)

**Pull request closed without merging**